### PR TITLE
refactor: use external nft descriptor

### PR DESCRIPTION
**Pull request by 0xsambugs**
We took the DCATokenDescriptor and moved it to a new repository (https://github.com/Mean-Finance/nft-descriptors)

This allows us to keep the core here, and add future NFT descriptor without modifying this repository

### CODE REVIEW

File: contracts/DCAPermissionsManager/DCAPermissionsManager.sol [hunk 1]
```diff
@@ -31,7 +30,7 @@ contract DCAPermissionsManager is ERC721, EIP712, Governable, IDCAPermissionMana
   /// @inheritdoc IDCAPermissionManager
   bytes32 public constant PERMISSION_SET_TYPEHASH = keccak256('PermissionSet(address operator,uint8[] permissions)');
   /// @inheritdoc IDCAPermissionManager
-  IDCATokenDescriptor public nftDescriptor;
```
Comment (0xsambugs) [inline contracts/DCAPermissionsManager/DCAPermissionsManager.sol:34]: It has the same interface, just renamed it

File: deploy/006_timelock.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
     },
     log: !process.env.TEST,
     overrides: {
-      gasLimit: 2_000_000,
```
Comment (0xsambugs) [inline deploy/006_timelock.ts:26]: I hate you so much gas limit!

File: test/integration/DCAHubPositionDescriptor/dca-hub-position-descriptor.spec.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import { expect } from 'chai';
+import { buildSwapInput } from 'js-lib/swap-utils';
+import { DeterministicFactory, DeterministicFactory__factory } from '@mean-finance/deterministic-factory/typechained';
+
+const WETH_ADDRESS = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2';
+const USDC_ADDRESS = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
+const WETH_WHALE_ADDRESS = '0xf04a5cc80b1e94c69b48f5ee68a08cd2f09a7c3e';
+const USDC_WHALE_ADDRESS = '0xcffad3200574698b78f32232aa9d63eabd290703';
+
+describe('DCAHubPositionDescriptor', () => {
```
Comment (0xsambugs) [inline test/integration/DCAHubPositionDescriptor/dca-hub-position-descriptor.spec.ts:19]: We basically took the e2e test, and made it an integration test

**Review by 0xged — APPROVED** [review]
Nice! A lot of stuff getting removed from this repo 😈

**Review by nchamo — APPROVED** [review]
Damn, look at all that deleted code!

**Pull request merged**